### PR TITLE
feat: helmcharts to deploy Eclipse Che

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatolii Bazko <abazko@redhat.com>

### What does this PR do?
Introduce Helm charts to deploy Eclipse Che

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20552

### How to test this PR?

See helmcharts/README.md

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [ ] [OLM bundles are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: helmcharts/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+Minimal Kubernetes version is 1.19
+
+### Helm Versions
+
+Helm charts are only tested with Helm version 3.7.1
+
+### Helm Chart Install
+
+Create a Namespace and install the helm chart for Eclipse Che.
```
Comment (mmorhun) [inline helmcharts/README.md:23]:
```suggestion
Create a Namespace and install the helm chart for Eclipse Che Operator.
```

File: helmcharts/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+Helm charts to deploy [Eclipse Che](https://www.eclipse.org/che/)
+
+### Kubernetes Versions
+
+Minimal Kubernetes version is 1.19
+
+### Helm Versions
+
+Helm charts are only tested with Helm version 3.7.1
```
Comment (mmorhun) [inline helmcharts/README.md:19]: I think we should try and detect minimal version and put something like `3.x.y+` here.

File: helmcharts/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Eclipse Che Helm Charts
+
+- [Charts](#charts)
+  - [Kubernetes Versions](#kubernetes-versions)
+  - [Helm Versions](#helm-versions)
+  - [Helm Chart Install](#helm-chart-install)
```
Comment (mmorhun) [inline helmcharts/README.md:6]: `Helm Chart Installation` or `Install the Helm Chart`

File: helmcharts/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# Eclipse Che Helm Charts
+
+- [Charts](#charts)
+  - [Kubernetes Versions](#kubernetes-versions)
+  - [Helm Versions](#helm-versions)
```
Comment (mmorhun) [inline helmcharts/README.md:5]: `Required Helm version`?

File: helmcharts/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# Eclipse Che Helm Charts
+
+- [Charts](#charts)
+  - [Kubernetes Versions](#kubernetes-versions)
```
Comment (mmorhun) [inline helmcharts/README.md:4]: `Required Kubernetes version`?

File: helmcharts/README.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
+# Eclipse Che Helm Charts
+
+- [Charts](#charts)
```
Comment (mmorhun) [inline helmcharts/README.md:3]: To me none of the headlines was clear. Maybe we should have:
Title
\- Prerequisites (K8s & Helm versions)
\- Installation

File: helmcharts/values.yaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+#
+
+k8s:
```
Comment (AndrienkoAleksandr) [inline helmcharts/values.yaml:13]: Maybe we should also provide namespace with value 'eclipse-che' by default ?

Comment (tolusha) [reply helmcharts/values.yaml:13]: Namespace is handled by helmcharts by default:
`helm install --namespace <NAMESPACE> ... `

**Comment by tolusha** [comment]
@mmorhun 
Fixed

**Review by AndrienkoAleksandr — APPROVED** [review]

**Review by mmorhun — APPROVED** [review]

**Pull request merged**